src/jlfrontend.scm

```diff
@@ -140,7 +140,7 @@
 
 (define (toplevel-only-expr? e)
   (and (pair? e)
-       (or (memq (car e) '(toplevel line module import using export
+       (or (memq (car e) '(toplevel line module import using export public
```

Comment:
Oh right :+1:

Am I right in thinking this is more an optimization than a bugfix?

Reply:
I guess mostly, but it's also possible some code in later passes might not know how to handle it. That doesn't seem to be the case currently though.